### DISCUSSION TITLE
How To Render the Current Site Theme On My Home Page

### DISCUSSION THREAD

**Discussion by krchome** [OrchardCMS/OrchardCore#16840] [Q&A] [answered]
I started catching up on Orchard Core with the latest version 2.0.2 and created the CMS website following the instructions at :https://docs.orchardcore.net/en/latest/getting-started/

Although I faced no problems in creating the website in [ASP.NET](http://asp.net/) 8 with the latest OC nuget packages and could login with the super user details. 

It was then that I became clueless, when I picked up a theme and enabled and made it current, like the Clean Blog theme.  as in the screenshot below:

![image](https://github.com/user-attachments/assets/90b6b7cc-1a4e-4e95-aaf1-3d808326ff21)

**I am not sure how to run this theme unlike the older version that I videoed in** [asp.net](http://asp.net/) core 3.1 about 4 years ago which would change the theme and it would be displayed straight away with the click of the run button as in this video: 
[https://youtu.be/bacIbCrTsng?si=zEAIIHBrqy0gR1Ob]  (at about 08:40.00 )

I still tried to work my way through (without any easy to navigate documentation that I could  find giving this most logical next step), by clicking on the tenants as below screenshot: 

![image](https://github.com/user-attachments/assets/7c33cef8-c256-42d4-869d-4f20230e3e54)

This rendered the following home page : 
![image](https://github.com/user-attachments/assets/d036dcd6-2d96-4468-bbae-d09093b74fb9)

Although I was expecting the clean blog theme!

Long story short:  **Could  some one tell us, How Do we view the Current Theme Rendered on My Home Page ?**  

Best 
Kaushik Roy Chowdhury

**Comment by hishamco** [comment]
If your recipe has a Blank theme, you can customize the page. From the picture above I see TheBlogTheme is the active theme, that's why your page is themed

Create a new theme then make it as current, or you could add Razor Page inside `Pages` folder in your website and it will show up without a theme

**Reply by krchome** [reply]
Hi @hishamco, 
Your comments "If your recipe has a Blank theme, you can customize the page" **How do I do it ?** **Also Why do I need to have to do it in the first place ?** 
If you see my video link in the discussion you'll find it was the exact same process leading to the creation of the admin theme (with empty ASP.NET core web app, yet thereafter changing and restoring the current theme like Agency theme (or others) was the result of clicking of the "Run" button. Just one button click and your site will take the new theme. 
Also "Create a new theme then make it as current," Again the question is **"What are the steps to follow?"** 

I feel the documentation needs to be thoroughly updated keeping in view the changes with moderation by concerned people.

My take is that packing the latest version with many extra features, though the things that were achievable to any user in an earlier version are not apparent, only compromises the usability of the great platform. Needless to say for any UI the usability should be at the foremost for everyone !

Hope I shall get a suitable response this time to address all my concerns .

Kaushik

**Comment by hishamco** [comment]
> Your comments "If your recipe has a Blank theme, you can customize the page" How do I do it ? Also Why do I need to have to do it in the first place ?

You can choose your default theme during the setup. My question again did you create a new theme?

**Reply by krchome** [reply, marked as answer]
Hi @hishamco 
Updates : Now you were right in saying that I picked up SAAS earlier in the theme/recipes dropdown. So I redid everything from scratch and chose Agency theme and it worked all right with Ageny theme appearing. Thanks very much and my bad  :-).

On a side note : Once I select a theme and then try to change it to another once say from initial Agency to Clean Blog. The changed theme doesn't seem to appear in the way that you'd expect. 
Can you shed some light about changing from one theme to another ? Is it not something that is in order?

Cheers,
Kaushik

**Reply by hishamco** [reply]
You can change the theme after setup from the `Design -> Themes` then the page will inherit the style from the current theme

**Comment by krchome** [comment]
I have updated my comments on GitHub. Pls have a look.

Best.
Kaushik

**Comment by krchome** [comment]
As I said It allows me to change the theme as you suggested, but the
changed theme doesn't look what it should look! For example I created the
Agency theme initially from the setup for the first time. Then changed it
from Design - - Themes to Clean Blog. The resulting Clean Blog theme showed
up skewed. I then changed it a second time from Clean Blog to Coming Soon.
This time the page was rendered with a weird image but certainly not the
coming soon theme image. What is the story behind it any ideas? Since I am
responding from my mobile I can't paste the screenshots but you can try it
yourself to confirm it!

**Reply by hishamco** [reply]
Switching between the themes will not render the page with a proper style. It depends on the markup. I'm suggesting to add Liquid or Razor Pages to your theme, then they should be displayed properly for newly added theme

**Reply by ns8482e** [reply]
@krchome  theme can provide design elements like - colors, css, layout, images. 

 But it can also provide content elements such has content items, content parts, contents etc during setup viz recipe. - If the Theme contains setup recipe then it can provide content elements needed for setup during the setup process. Theme author can control visibility of recipes in "Recipes" menu

**Comment by krchome** [comment]
Thanks and a more detailed procedure of  adding Liquid or Razor Pages will
be appreciated. Can you please guide me in that direction?

Kaushik
